include/tcframe/generator/TestCaseGenerator.hpp

```diff
@@ -61,7 +61,9 @@ class TestCaseGenerator {
             applyInput(testCase);
             verifyInput(testCase);
             generateInput(testCase, inputFilename, config);
-            evaluateAndApplyOutput(testCase, inputFilename, outputFilename, config);
+            if (config.generateOutput()) {
```

Comment:
You missed this check in `Generator.combine()` (for multiple test cases in a file).